buildkitd/docker-auto-install.sh

```diff
@@ -2,4 +2,111 @@
 
 set -eu
 
-apk add --update --no-cache docker-compose
+distro=$(sed -n -e 's/^ID=\(.*\)/\1/p' /etc/os-release)
+
+detect_dockerd() {
+    set +e
+    command -v dockerd
+    has_d="$?"
+    set -e
+    return "$has_d"
+}
+
+detect_docker_compose() {
+    set +e
+    command -v docker-compose
+    has_dc="$?"
+    set -e
+    return "$has_dc"
+}
+
+print_debug() {
+    set +u
+    if [ "$EARTHLY_DEBUG" = "true" ] ; then
+        echo "$@"
+    fi
+    set -u
+}
+
+install_docker_compose() {
+    case "$distro" in
+        alpine)
+            apk add --update --no-cache docker-compose
+            ;;
+
+        *)
+            curl -L "https://github.com/docker/compose/releases/download/1.27.4/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+            chmod +x /usr/local/bin/docker-compose
+            ;;
+    esac
+}
+
+install_dockerd() {
+    case "$distro" in
+        alpine)
+            apk add --update --no-cache docker
+            ;;
+
+        ubuntu)
+            install_dockerd_debian_like
+            ;;
+
+        debian)
+            install_dockerd_debian_like
+            ;;
+
+        *)
+            echo "Warning: Distribution $distro not yet supported for Docker-in-Earthly."
+            echo "Will attempt to treat like Debian."
```

Comment:
How about expanding this to request the user creates a ticket under https://github.com/earthly/earthly/issues for additional OS support?

Reply:
Good idea